java-checks/src/main/java/org/sonar/java/checks/RedundantModifierCheck.java

```diff
@@ -64,6 +64,13 @@ public void visitNode(Tree tree) {
         checkRedundantModifier(modifiers, Modifier.FINAL);
       } else if(member.is(Kind.CONSTRUCTOR) && tree.is(Kind.ENUM)) {
         checkRedundantModifier(((MethodTree) member).modifiers(), Modifier.PRIVATE);
+      } else if (member.is(Kind.CLASS, Kind.INTERFACE) && isInterfaceOrAnnotation(tree)) {
+        ClassTree nested = (ClassTree) member;
+        ModifiersTree modifiers = nested.modifiers();
+        checkRedundantModifier(modifiers, Modifier.PUBLIC);
+        checkRedundantModifier(modifiers, Modifier.STATIC);
+      } else if (member.is(Kind.INTERFACE) && tree.is(Kind.CLASS, Kind.ENUM)) {
+        checkRedundantModifier(((ClassTree) member).modifiers(), Modifier.STATIC);
```

Comment:
In general I'll suggest to refactor it `switch {}` to have sth like this:

`      switch (member.kind()) {
        case METHOD:
          checkMethod(classTree, (MethodTree) member);
          break;
        case VARIABLE:
          checkVariable(classTree, (VariableTree) member);
          break;
        case CONSTRUCTOR: ...
        case CLASS: ...
        case INTERFACE: ...

      }`
To my mind it will be more readable, too many else if is quite difficult to understand